darts/tests/models/forecasting/test_regression_models.py

```diff
@@ -649,71 +743,176 @@ def test_static_covs_addition(self):
+            target_series, past_covs, future_covs, max_samples_per_ts=None
         )[0]
-        self.assertEqual(
-            features.shape,
-            (max_samples * len(all_series), all_series_width + max_scovs_width),
-        )
+        self.assertEqual(features.shape, (expected_height, expected_width))
 
         # no static covs at prediction but static covs at training
-        reg_model.fit(all_series)
+        reg_model.fit(target_series)
         pred_features = reg_model._create_lagged_data(
-            series4, None, None, max_samples_per_ts=1
+            series_no_statics, past_covs, future_covs, max_samples_per_ts=1
         )[
             0
         ]  # simulates features prep at prediction time
-        self.assertEqual(pred_features.shape, (1, all_series_width + max_scovs_width))
+        self.assertEqual(pred_features.shape, (1, expected_width))
+
+        # different sizes of past and future covariates + different lenghts of target series
+        target_lag = [-2, -1]
+        past_covs_lag = [-3]
+        future_covs_lag = [-1, 3]
+        output_chunk_length = 4
+
+        target_series = [series1, series2, series3]
+        past_covs = [series2, series1, series3]
+        future_covs = [series3, series3, series1]
+
+        reg_model = RegressionModel(
+            lags=target_lag,
+            lags_past_covariates=past_covs_lag,
+            lags_future_covariates=future_covs_lag,
+            output_chunk_length=output_chunk_length,
+        )
+
+        expected_height, expected_width = self.helper_get_expected_shape_static_covs(
+            reg_model,
+            target_series,
+            past_covs,
+            future_covs,
+            target_lag,
+            past_covs_lag,
+            future_covs_lag,
+        )
+
+        features = reg_model._create_lagged_data(
+            target_series, past_covs, future_covs, max_samples_per_ts=None
+        )[0]
+        self.assertEqual(features.shape, (expected_height, expected_width))
+
+        # outptut_chunk_length < max_future_cov_lag and len(target_series) = len(future_covs)
+        target_lag = [-1]
+        past_covs_lag = None
+        future_covs_lag = [-1, 3]
+        output_chunk_length = 2
+
+        target_series = [series1, series2, series3]
+        past_covs = None
+        future_covs = [series1, series2, series3]
+
+        reg_model = RegressionModel(
+            lags=target_lag,
+            lags_past_covariates=past_covs_lag,
+            lags_future_covariates=future_covs_lag,
+            output_chunk_length=output_chunk_length,
+        )
+
+        expected_height, expected_width = self.helper_get_expected_shape_static_covs(
+            reg_model,
+            target_series,
+            past_covs,
+            future_covs,
+            target_lag,
+            past_covs_lag,
+            future_covs_lag,
+        )
+        features = reg_model._create_lagged_data(
```

Comment:
In these tests you test `_create_lagged_data` directly, which is good. Could we also maybe add another "higher level" test going through all the fit/predict process with one model in a difficult case (series of varying lengths and widths with static covs) and testing that the final outputs are as expected?

Reply:
ok will do